Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp

```diff
@@ -719,6 +741,17 @@ void WebAutomationSession::willShowJavaScriptDialog(WebPageProxy& page)
         if (!page->hasRunningProcess() || !m_client || !m_client->isShowingJavaScriptDialogOnPage(*this, page))
             return;
 
+#if ENABLE(WEBDRIVER_BIDI)
+        String dialogMessage = m_client->messageOfCurrentJavaScriptDialogOnPage(*this, page);
+        String dialogDefaultText = m_client->defaultTextOfCurrentJavaScriptDialogOnPage(*this, page);
+        std::optional<API::AutomationSessionClient::JavaScriptDialogType> apiDialogType = m_client->typeOfCurrentJavaScriptDialogOnPage(*this, page);
+        Inspector::Protocol::BidiBrowsingContext::UserPromptType userPromptType = toProtocolUserPromptType(apiDialogType.value_or(API::AutomationSessionClient::JavaScriptDialogType::Prompt));
```

Comment:
NIT: i feel like both of these could/should be `auto`

Reply:
I'll try that again, I recall it not being able to infer the optional type at least.